src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -125,38 +146,68 @@ public class PullRequestCreator
                         description,
                         originBranch,
                         UpstreamBranch,
-                        maintainersCanModify);
+                        MaintainersCanModify);
                 }
             }
         }
 
+        private string CreateDiscardedCommitListBody(
+            GitHubPullRequest pullRequestToUpdate,
+            GitCommit oldCommit,
+            GitHubClient client)
+        {
+
+            // GitHub returns the HTML "commit" url, but we want "commits" so that CI results show.
+            string oldCommitsUrl = oldCommit.HtmlUrl.Replace("commit", "commits");
+
+            string oldCommitLine =
+                $" * [`{oldCommit.Sha.Substring(0, 7)}`]({oldCommitsUrl}) {oldCommit.Message}\r\n";
+
+            // Find insertion point. GitHub always returns \r\n.
+            string insertionMarker = $"<{DiscardedCommitElementName}>\r\n\r\n";
+            string endInsertionMarker = $"\r\n</{DiscardedCommitElementName}>";
+
+            int elementBegin = pullRequestToUpdate.Body.IndexOf(
+                insertionMarker,
+                StringComparison.Ordinal);
+
+            if (elementBegin != -1)
+            {
+                return pullRequestToUpdate.Body.Insert(
+                    elementBegin + insertionMarker.Length,
+                    oldCommitLine);
+            }
+            return pullRequestToUpdate.Body +
```

Comment:
I'm a bit confused here.  I would have thought the new commit would have just applied on top of the old commit.  But it appears here that the old commits just get appended to the new commit's message.

What's the user scenario for "TrackDiscardedCommits"?  Do people want to be able to see the CI run details for each separate commit?  Or do they just want a list of them?

Reply:
I described the problem with putting the new commit on top here: https://github.com/dotnet/buildtools/issues/1727#issuecomment-337669300.

This is the PR description, not the commit message, if that clarifies anything.

The end goal is to see CI run details and source code changes, and the links go to the page where you can get to both. This isn't the best workflow (a few extra clicks), so a future improvement would be to include the CI statuses in the list entry for each discarded commit.